tracks/julia/exercises/nucleotide-count/mentoring.md

```diff
@@ -92,4 +86,15 @@ function count_nucleotides4(strand)
     result
 end
 ```
+
+Its speed comes from three sources:
+
+1. Iterating a vector of bytes instead of characters (though this algorithm is still unicode-safe).
+   This is faster because iterating characters from a string involves scanning the same vector of bytes but also doing a little work every iteration to establish if the current byte is the start of a multibyte sequence (and, if so to reassemble that sequence into its proper 32 bit character).
+2. The loop body is extremely simple and fast on x86 processors (a load, an addition and a store) and branchless (there are no conditionals within the loop body).
+   Branchless code is often faster because branching hinders CPU instruction level parallelism and stops compilers using "Single Instruction Multiple Data" (SIMD) instructions (though this loop body can't benefit from SIMD anyway).
```

Comment:
Perhaps add a note on how one can see that using `@code_native`?

Reply:
A dangerous thing to ask when I'm procrastinating.